test/ctia/test_helpers/core.clj

```diff
@@ -290,59 +307,71 @@
                                                   :role role
                                                   :capabilities caps})))
 
-(defmacro deftest-for-each-fixture [test-name fixture-map & body]
+(defmacro deftest-for-each-fixture-with-app [test-name fixture-map app & body]
+  (assert (simple-symbol? app) (pr-str app))
   `(do
      ~@(for [[name-key fixture-fn] fixture-map]
          `(test/deftest ~(with-meta (symbol (str test-name "-" (name name-key)))
                                     {(keyword name-key) true})
-            (~fixture-fn (fn [] (do ~@body)))))))
+            (~fixture-fn
+              (fn []
+                (let [~app (get-current-app)]
+                  ~@body)))))))
 
-(defn get-http-port []
-  ;; Note: using current-get-in-config-fn here is a hack and should be refactored
-  ;; to function parameter once the initial TK setup is merged into master.
-  ((current-get-in-config-fn) [:ctia :http :port]))
+(defn get-http-port [app]
```

Comment:
`:- s/Int`, so that things fail-fast instead of at http level later

Reply:
The 'feedback' commit below had the wrong `:-` syntax, fixed in the `fix spec return syntax` commit.